Snowflake.Data.Tests/FastMemoryStreamTest.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Core;
+    using System.Linq;
+
+    [TestFixture]
+    class FastMemoryStreamTest
+    {
+        FastMemoryStream fastMemoryStream;
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            fastMemoryStream = new FastMemoryStream();
+        }
+
+        [Test]
+        [Ignore("FastMemoryStreamTest")]
+        public void FastMemoryStreamTestDone()
+        {
+            // Do nothing;
+        }
+
+        [Test]
+        public void TestDefaultValues()
+        {
+            // FastMemoryStream defaults to size of 0 and buffer of size 256
+            Assert.AreEqual(0, fastMemoryStream.Length);
+            Assert.AreEqual(256, fastMemoryStream.GetBuffer().Length);
+        }
+
+        [Test]
+        public void TestWriteByte()
+        {
+            byte val = 32;
+            int bufferLength = fastMemoryStream.GetBuffer().Length;
+
+            byte[] byteArr = Enumerable.Repeat(val, bufferLength).ToArray();
+
+            for (int i = 0; i < bufferLength; i++)
+            {
+                fastMemoryStream.WriteByte(val);
+            }
+
+            Assert.AreEqual(byteArr, fastMemoryStream.GetBuffer());
+        }
+
+        [Test]
+        public void TestClear()
+        {
+            fastMemoryStream.Clear();
```

Comment:
should we write anything to the stream before to test if clear method clears the stream?

Reply:
I'll add that to the test